src/modm/platform/uart/rtt/rtt.cpp.in

```diff
@@ -66,7 +66,8 @@ static uint8_t rx_data_buffer_{{loop.index0}}[{{size}}];
 %#
 struct RttControlBlock
 {
-	const char identifier[16]{"modm.rtt.modm"};
+	RttControlBlock() { ((uint32_t*)identifier)[1] = 0x7474722e; } // .rtt
+	char identifier[16]{"modm    .modm"};
```

Comment:
Can't we do a cleaner solution without [UB](https://en.cppreference.com/w/cpp/language/reinterpret_cast#Type_aliasing)?
I could imagine removing the `static` on the declaration of `rtt_control` in the same file would force it to put the structure into memory completely because it can't conclude anymore the array is not accessed from another translation unit.

Reply:
I'm actually more annoyed that this is not a POD type and therefore is constructed at runtime with the constructor. I was hoping for a simple copy from the Flash memory as part of the .data section… Any clue how to get that done? Is it the volatile members?

Reply:
And yes: memcpy would be cleaner and also works.

Reply:
> I'm actually more annoyed that this is not a POD type and therefore is constructed at runtime with the constructor. I was hoping for a simple copy from the Flash memory as part of the .data section… Any clue how to get that done?

POD means standard layout and trivial. As long as construction does something, the type can't be trivial in the language sense. Any non-static data member with initializer is enough to make it non trivially-constructible, even a const one suffices.

But I don't think the compiler strictly requires that to copy it from flash to static storage. Probably the user-defined constructor is the culprit. To enforce no constructor code runs at start-up, declare the constructor as `constexpr` and the variable as `constinit`. In that case you can't reinterpret_cast anything because it is not allowed in compile time evaluation.

Reply:
`volatile` members don't work with `constinit` though ...

Reply:
Awesome hint! `constinit` is **exactly** what I needed, it forces the initialization into the `.data` section because the compiler *must* prevent a call to a compiler, otherwise there's a possibility for static initialization order fiasco.

We could probably declare most objects in the modm library as constinit (if the class supports it of course), since we explicitly deal with init order via `.hardware_init`. It could save us some flash space, since the constructing the data via instructions is fairly expensive. But it's just an optimization of course, I'll put in on my ToDo list.